.github/workflows/run-tests.yml

```
@@ -0,0 +1,72 @@
+name: Tests
+
+on: [push, pull_request]
```

I'm fairly certain this is why there are double failures.

---

```suggestion
on:
  pull_request:
    branches:    
      - master
```

This would run on PR to master. When combined with branch protection rules, this can then enable all the benefits of CI, without worrying about pushes. Pretty sure it also runs when you push to the PR.

---

I don't think this is what makes the actions run twice. I've already tested this and use this many other repositories.

---

This is what I've seen in an other proejct: https://github.com/nunomaduro/larastan/blob/master/.github/workflows/tests.yml#L3

The only thing it prevents, if running a job twice when doing a push on the main repository, but not on the main branch.

---

If that might help ...
You should do something like that:
```suggestion
on:
  push:
    branches: [master]
  pull_request:
    types: [opened, synchronize, reopened]
  release:
    types: [created]
```

That way, the workflow is triggered for pull requests, even for external pull requests.
The `push` event is only triggered when there is a merge/commit in `master` branch, and not in the PR branch.
The `release` event is optional, it helps trigger the workflow when a release is created.